geospaas_adas_viewer/tests.py

```diff
@@ -0,0 +1,125 @@
+        # The second one should not be in the html
+        self.assertFalse(any([('file://localhost/some/test/file2.ext' in dat)
+                              for dat in self.parser.data]))
+
+    def test_the_post_verb_of_GUI_with_common_part_of_parameter_name(self):
+        """shall return the uri of fixtures' datasets in the specified placement
+        of datasets inside the resulted HTML in the case of a POST request without
+        any polygon from user """
+        res4 = self.client.post('/adas/', {
+            'time_coverage_start': timezone.datetime(2000, 12, 29),
+            'time_coverage_end': timezone.datetime(2020, 1, 1),
+            'source': 1,
+            'nameparameters': 'ame'})  # <= Notice the part of both names of 'name_fake' and 'dummy_name' provided by user
+        self.assertEqual(res4.status_code, 200)
+        self.parser.feed(str(res4.content))
+        # both datasets must be in the html
+        self.assertTrue(any([('file://localhost/some/test/file1.ext' in dat)
+                             for dat in self.parser.data]))
+        self.assertTrue(any([('file://localhost/some/test/file2.ext' in dat)
+                             for dat in self.parser.data]))
+
+    def test_the_get_verb_of_adas_GUI(self):
+        """shall return ALL uri of fixtures' datasets in the specified placement
+        of datasets inside the resulted HTML in the case of a GET request"""
+        res5 = self.client.get('/adas/')
+        self.assertEqual(res5.status_code, 200)
+        self.parser.feed(str(res5.content))
+        # both datasets must be in the html
+        self.assertTrue(any([('file://localhost/some/test/file1.ext' in dat)
+                             for dat in self.parser.data]))
+        self.assertTrue(any([('file://localhost/some/test/file2.ext' in dat)
+                             for dat in self.parser.data]))
+
+
+class FilteringFunctionalityTestsForADASGUI(TestCase):
+    """ Unit tests for filter method which is placed inside the basic form """
+    fixtures = ["vocabularies", "catalog"]
+
+    def setUp(self):
+        # setting up fake parameters for fixture datasets
+        self.ds = Dataset.objects.all()
+        p1 = Parameter(short_name='dummy',
+                       standard_name='dummy_name', units='dummy')
+        p1.save()
+        self.ds[1].parameters.add(p1)
+        p2 = Parameter(short_name='dummy',
+                       standard_name='name_fake', units='dummy')
+        p2.save()
+        # p2 is added to 'NERSC_test_dataset_titusen' dataset with pk=1
+        self.ds[0].parameters.add(p2)
+
+    def tearDown(self):
+        self.form = None
+        self.ds = None
+
+    def test_filtering_functionality_by_a_fraction_of_parameter_name(self):
+        """shall return the first dataset in fixtures
+        since the 'fak' is set in the name of parameters of
+        the first one """
+        self.form = adasIndexView.form_class({'polygon': '',
```

Comment:
Neither form, nor ds should belong to self. Fix below also.
```suggestion
        form = adasIndexView.form_class({'polygon': '',
```

Reply:
kindly, for using setUp at the beginning of the test suit, we need to write them with self. OK?